lib/features/add_property/view/widgets/add_property_images_view.dart

```diff
@@ -13,28 +16,62 @@ class AddPropertyImagesView extends StatelessWidget {
   Widget build(BuildContext context) {
     return BlocProvider<AddPropertyImagesCubit>(
       create: (context) => AddPropertyImagesCubit(),
-      child: const AddPropertyImagesContent(),
+      child: const AddPropertyImagesBody(),
     );
   }
 }
 
-class AddPropertyImagesContent extends StatelessWidget {
-  const AddPropertyImagesContent({Key? key}) : super(key: key);
+class AddPropertyImagesBody extends StatefulWidget {
+  const AddPropertyImagesBody({Key? key}) : super(key: key);
+
+  @override
+  AddPropertyImagesBodyState createState() => AddPropertyImagesBodyState();
+}
+
+class AddPropertyImagesBodyState extends State<AddPropertyImagesBody>
+    with WidgetsBindingObserver {
+  @override
+  void initState() {
+    super.initState();
+    WidgetsBinding.instance.addObserver(this);
+  }
+
+  @override
+  void dispose() {
+    WidgetsBinding.instance.removeObserver(this);
+    super.dispose();
+  }
+
+  @override
+  void didChangeAppLifecycleState(AppLifecycleState state) {
+    if (state == AppLifecycleState.resumed) {
+      context.read<AddPropertyImagesCubit>().screenResumed();
+    }
+  }
 
   @override
   Widget build(BuildContext context) {
     return BlocListener<AddPropertyImagesCubit, AddPropertyImagesState>(
-      listener: (context, state) {
+      listener: (_, state) {
+        if (state is PhotoPermissionGranted) {
+          _showSelectPhotoBottomSheet(context).then((isShown) {
+            context
+                .read<AddPropertyImagesCubit>()
+                .dismissSelectPhotoPermissionBottomSheet(isShown);
+          });
+          context.read<AddPropertyImagesCubit>().openSelectPhotoBottomSheet();
```

Comment:
leave this to the Select photo bottom sheet, this event will be fired by Select photo bottom sheet

Reply:
already removed